src/osd/PG.cc

```diff
@@ -3298,7 +3298,11 @@ void PG::reg_next_scrub()
       (info.stats.stats_invalid && g_conf->osd_scrub_invalid_stats)) {
     scrubber.scrub_reg_stamp = utime_t();
   } else {
-    scrubber.scrub_reg_stamp = info.history.last_scrub_stamp;
+    if (info.history.last_scrub_stamp + cct->_conf->osd_scrub_min_interval <
+        info.history.last_deep_scrub_stamp + cct->_conf->osd_deep_scrub_interval)
+      scrubber.scrub_reg_stamp = info.history.last_scrub_stamp;
+    else
+      scrubber.scrub_reg_stamp = info.history.last_deep_scrub_stamp;
```

Comment:
Let's say it has been 1/2 day since last regular scrub and 8 days since last deep scrub.   The change to reg_next_scrub() will indeed cause that pg to come up earlier in OSD::sched_scrub().  However, it will NOT call PG::sched_scrub() until at least the min interval since the last scrub.  As your own code change to OSD::sched_scrub() has to replace the pos.first with last_scrub_stamp because that logic only cares of normal scrub interval.  This prevents a deep scrub starting too soon after the last regular scrub.